plugins/node/opentelemetry-instrumentation-mysql/src/instrumentation.ts

```diff
@@ -343,4 +384,56 @@ export class MySQLInstrumentation extends InstrumentationBase<
       };
     };
   }
+  private _setPoolcallbacks(
+    pool: mysqlTypes.Pool,
+    thisPlugin: MySQLInstrumentation,
+    id: string
+  ) {
+    //TODO:: use semantic convention
+    const poolName = id || getPoolName(pool);
+
+    pool.on('connection', connection => {
+      thisPlugin._connectionsUsage.add(1, {
+        state: 'idle',
+        name: poolName,
+      });
+      connection.on('end', () => {
+        thisPlugin._connectionsUsage.add(-1, {
+          state: 'idle',
+          name: poolName,
+        });
+      });
+    });
+
+    pool.on('acquire', connection => {
+      thisPlugin._connectionsUsage.add(-1, {
+        state: 'idle',
+        name: poolName,
+      });
+      thisPlugin._connectionsUsage.add(1, {
+        state: 'used',
+        name: poolName,
+      });
+    });
+
+    pool.on('release', connection => {
+      thisPlugin._connectionsUsage.add(-1, {
+        state: 'used',
+        name: poolName,
+      });
+      thisPlugin._connectionsUsage.add(1, {
+        state: 'idle',
+        name: poolName,
+      });
+    });
+  }
+}
+function getPoolName(pool: mysqlTypes.Pool): string {
+  const c = pool.config.connectionConfig;
+  let poolName = '';
+  poolName += c.host ? `host: ${c.host} ` : '';
+  poolName += c.port ? `port: ${c.port} ` : '';
+  poolName += c.database ? `database: ${c.database} ` : '';
+  poolName += c.user ? `user: ${c.user}` : '';
+  return poolName;
```

Comment:
This might have a trailing space if there is no user. Is the pool name specified in the semconv or is this a scheme you came up with?

Reply:
I'll trim the string. 
the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/database-metrics.md) doesn't say how the `pool.name` should look like, this is why I opened this issue: https://github.com/open-telemetry/opentelemetry-specification/issues/2958 and you can see there Mateusz's answer. 
If there is a need to put it in the semconv first, please let me know.

Reply:
I moved 'getPoolName' to the utils, and add a test file named: 'utils.test.ts' that test this function.